File: Map/Z_InterpretorWCB/src/domain/ExeStack.java
package domain;

import java.util.ArrayDeque;
import java.util.Deque;

public class ExeStack<T> implements IStack<T>
{
	private Deque<T> sk;
	
	public ExeStack()
	{
		sk  = new ArrayDeque<T>();
	}

	@Override
	public T pop() 
	{
		return sk.pop();
	}

	@Override
	public void push(T value) 
	{
		sk.push(value);
	}

	@Override
	public boolean isEmpty() 
	{
		return sk.isEmpty();
	}

	@Override
	public T first() 
	{
		return sk.peek();
	}
	
	@Override
	public String toString()
	{
		StringBuffer sb = new StringBuffer();
		for(T e : sk)
			sb.append(""+e+", ");
		return sb.toString();
	}

	@Override
	public Iterable<T> getAll() 
	{
		return sk;
	}


}

File: Map/Map/Map/domain/ExeStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Map.domain
{
    public class ExeStack<T> : IStack<T>
    {
        private Stack<T> stack;

        public ExeStack()
        {
            stack = new Stack<T>();
        }

        public IEnumerable<T> getAll()
        {
            return stack.AsEnumerable<T>();
        }

        public bool isEmpty()
        {
            return stack.Count == 0;
        }

        public T pop()
        {
            return stack.Pop();
        }

        public void push(T value)
        {
            stack.Push(value);
        }

        public override String ToString()
        {
            StringBuilder sb = new StringBuilder();
            foreach(T x in stack)
            {
                sb.Append(x + ", ");
            }
            return sb.ToString();
        }
    }
}
